sdk/storage/storage-file-datalake/review/storage-file-datalake.api.md

```diff
@@ -0,0 +1,1424 @@
+    // (undocumented)
+    abortSignal?: AbortSignalLike;
+    // (undocumented)
+    conditions?: DataLakeRequestConditions;
+}
+
+// @public (undocumented)
+export type PathSetHttpHeadersResponse = PathSetHttpHeadersHeaders & {
+    _response: HttpResponse & {
+        parsedHeaders: PathSetHttpHeadersHeaders;
+    };
+};
+
+// @public (undocumented)
+export interface PathSetMetadataHeaders {
+    // (undocumented)
+    clientRequestId?: string;
+    // (undocumented)
+    date?: Date;
+    // (undocumented)
+    encryptionKeySha256?: string;
+    // (undocumented)
+    etag?: string;
+    // (undocumented)
+    isServerEncrypted?: boolean;
+    // (undocumented)
+    lastModified?: Date;
+    // (undocumented)
+    requestId?: string;
+    // (undocumented)
+    version?: string;
+}
+
+// @public (undocumented)
+export interface PathSetMetadataOptions extends CommonOptions {
+    // (undocumented)
+    abortSignal?: AbortSignalLike;
+    // (undocumented)
+    conditions?: DataLakeRequestConditions;
+}
+
+// @public (undocumented)
+export type PathSetMetadataResponse = PathSetMetadataHeaders & {
+    _response: HttpResponse & {
+        parsedHeaders: PathSetMetadataHeaders;
+    };
+};
+
+// @public (undocumented)
+export interface PathSetPermissionsOptions extends CommonOptions {
+    // (undocumented)
+    abortSignal?: AbortSignalLike;
+    // (undocumented)
+    conditions?: DataLakeRequestConditions;
+    // (undocumented)
+    group?: string;
+    // (undocumented)
+    owner?: string;
+}
+
+// Warning: (ae-forgotten-export) The symbol "PathUpdateHeaders" needs to be exported by the entry point index.d.ts
+//
+// @public
+type PathUpdateResponse = PathUpdateHeaders & {
+    _response: coreHttp.HttpResponse & {
+        parsedHeaders: PathUpdateHeaders;
+    };
+};
+
+export { PathUpdateResponse as FileAppendResponse }
+
+export { PathUpdateResponse as FileFlushResponse }
+
+// @public
+export class Pipeline extends Pipeline_2 {
+    constructor(factories: RequestPolicyFactory[], options?: PipelineOptions);
+    readonly factories: RequestPolicyFactory[];
+    readonly options: PipelineOptions;
+    toServiceClientOptions(): ServiceClientOptions;
+}
+
+// @public
+export interface PipelineOptions {
+    httpClient?: IHttpClient;
+}
+
+// @public (undocumented)
+export type PublicAccessType = "filesystem" | "file";
+
+// @public (undocumented)
+export interface RawAccessPolicy {
+    // (undocumented)
+    expiresOn?: string;
+    // (undocumented)
+    permissions: string;
+    // (undocumented)
+    startsOn?: string;
+}
+
+export { RequestPolicy }
+
+export { RequestPolicyFactory }
+
+export { RequestPolicyOptions }
+
+export { RestError }
+
+// @public (undocumented)
+export interface RolePermissions {
+    // (undocumented)
+    execute: boolean;
+    // (undocumented)
+    read: boolean;
+    // (undocumented)
+    write: boolean;
+}
+
+// @public
+export interface SasIPRange {
```

Comment:
In hindsight I'm sad about the different casing of SAS here, but since it's this way in blobs as well, we shouldn't change it 😥

Reply:
Yeah. It's a unluckly missing in blob GA. We can update in next major release.